src/core/account_builder.cpp

```diff
@@ -0,0 +1,278 @@
+        }
+        catch (const chocan_user_exception &err)
+        {
+
+            issues.emplace(err);
+        }
+    }
+    else if (current_phase == building_phase.street_address)
+    {
+
+        parseAddress(input);
+
+        try
+        {
+            zip = std::stol(account_field.at(current_zip));
+        }
+        catch (const std::invalid_argument &err)
+        {
+            zip = 0;
+        }
+        catch (const std::out_of_range &err)
+        {
+            zip = 0;
+        }
+
+        try
+        {
+            address = Address(account_field.at(current_street), account_field.at(current_city), account_field.at(current_state), zip.value());
+        }
+        catch (const chocan_user_exception &err)
+        {
+            issues.emplace(err);
+
+        }
+    }
+
+    if (!issues)
+        build_phase.pop();
+
+    return;
+}
+
+const chocan_user_exception Account_Builder::get_current_issues()
+{
+
+    std::optional<chocan_user_exception> return_msg = issues;
+
+    issues.reset();
+
+    return return_msg.value_or(chocan_user_exception{"", {}});
+}
+
+const std::string Account_Builder::get_current_field() const
+{
+
+    if (build_phase.empty())
+        return review_account();
+
+    return build_phase.top();
+}
+
+void Account_Builder::parseName(const std::string &input)
+{
+
+    size_t delimiter_position = input.find(",");
+
+    if(delimiter_position == std::string::npos){
+    
+        account_field[current_first] = "";
+        account_field[current_last] = "";
+
+    }else{
+
+        account_field[current_first] = input.substr(0, input.find(","));
+        account_field[current_last] = input.substr(input.find(",") + 1, input.length());
+
+        remove_leading_white_space(account_field[current_first]);
+        remove_leading_white_space(account_field[current_last]);
+    }
+
+}
+
+void Account_Builder::parseAddress(const std::string &input)
+{
+
+    size_t delimiter_position = 0;
+    const std::string delimiter = ",";
+    std::string temp_input(input + delimiter);
+    std::stack<field_types> current_field({current_zip, current_state, current_city, current_street});
+
+    while (!current_field.empty() && (delimiter_position = temp_input.find(delimiter)) != std::string::npos)
+    {
+        account_field[current_field.top()] = temp_input.substr(0, delimiter_position);
+
+        remove_leading_white_space(account_field[current_field.top()]);
+
+        current_field.pop();
+
+        temp_input.erase(0, delimiter_position + delimiter.length());
+    }
+
+    while(!current_field.empty()){
+        account_field[current_field.top()] = "";
+        current_field.pop();
+    }
+
+    account_field[current_zip] = account_field[current_zip].substr(0,5);
+}
+
+const std::string Account_Builder::valid_account_type(const std::string &input)
+{
+
+    std::string temp("");
+
+    for (std::vector<std::string>::iterator it = valid_types.begin(); it <= valid_types.end(); ++it)
```

Comment:
Maybe change valid_types to a set and use it's .find() function? Also, maybe just lowercase the entire input string and compare it the strings "member", "provider", "manager". That way we won't care what casing the input string is in and we won't need to compare it against the upper cased versions.

Reply:
i capitilize the input and check it against the three options now.